src/common/config.cpp

```diff
@@ -325,9 +325,19 @@ void setMainWindowGeometry(u32 x, u32 y, u32 w, u32 h) {
     main_window_geometry_w = w;
     main_window_geometry_h = h;
 }
-void setGameInstallDirs(const std::vector<std::filesystem::path>& dir) {
-    settings_install_dirs.resize(dir.size());
-    settings_install_dirs = dir;
+bool addGameInstallDir(const std::filesystem::path& dir) {
+    if (std::find(settings_install_dirs.begin(), settings_install_dirs.end(), dir) ==
+        settings_install_dirs.end()) {
+        settings_install_dirs.push_back(dir);
+        return true;
+    }
+    return false;
+}
```

Comment:
This should be changed such that its only called from addFolderButton since its the only place from where we add folders. Rather a check should be added to check if "installDir" has a path, if so then adding just installDir's path to installDirs then deleting installDir, such that we migrate away from calling installDir. Also, we will avoid spending time checking if a game folder exists within the game folders we saved before the current session.

Reply:
And add a deprecation notice for the migrating code. (possibly after 1 major release)

Reply:
There's already a migration function in config load, if you're talking about there not being one yet.

Reply:
Like i said we just need to add a comment saying that the code needs to be deprecated and i didn't say that there wasn't one. So that we remember to remove the code after a major release or so.

Reply:
> Like i said we just need to add a comment saying that the code needs to be deprecated and i didn't say that there wasn't one. So that we remember to remove the code after a major release or so.

Is this what you suggest? https://github.com/shadps4-emu/shadPS4/pull/1329/commits/6102470fa9adcd2675a55fb0542f47e00fd97857